src/TodayEventMonitor.vala

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2021 elementary, Inc. (https://elementary.io)
+ *           2014 Corentin Noël <corentin@elementary.io>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+public class Calendar.TodayEventMonitor : GLib.Object {
+    private Gee.HashMultiMap<ECal.Component, string> event_uids;
+
+    construct {
+        load_today_events ();
+        Timeout.add_seconds (86400, () => {
+            event_uids.clear ();
+            load_today_events ();
+            return true;
+        });
+    }
+
+    private void load_today_events () {
+        event_uids = new Gee.HashMultiMap<ECal.Component, string> ();
+        var model = new Calendar.EventStore ();
+        model.events_added.connect (on_events_added);
+        model.events_updated.connect (on_events_updated);
+        model.events_removed.connect (on_events_removed);
+        model.month_start = Calendar.Util.datetime_get_start_of_month (new DateTime.now_local ());
+    }
+
+    private void on_events_added (E.Source source, Gee.Collection<ECal.Component> events) {
+        var extension = (E.SourceAlarms)source.get_extension (E.SOURCE_EXTENSION_ALARMS);
+        if (extension.get_include_me () == false) {
+            return;
+        }
+
+        Idle.add ( () => {
+            foreach (var event in events)
+                add_event (source, event);
+
+            return false;
+        });
+    }
+
+    private void on_events_updated (E.Source source, Gee.Collection<ECal.Component> events) {
+        Idle.add ( () => {
+            foreach (var event in events)
+                update_event (source, event);
```

Comment:
Rather strangely, I'm seeing that when I edit an event the `events` collection has size 0, so the callback is never added. I'm not sure quite how to debug this, so if you have any suggestions, let me know. In the meantime I'll keep digging and update if I find anything.

Reply:
Never mind, I was seeing a combination of an EventStore bug and a user error about notifications. Ignore this.